src/NHibernate/Transaction/AdoTransaction.cs

```diff
@@ -108,7 +108,7 @@ public void Begin()
 		/// </exception>
 		public void Begin(IsolationLevel isolationLevel)
 		{
-			using (new SessionIdLoggingContext(sessionId))
+			using (session.BeginProcess())
```

Comment:
Here using the stored session id is not useful, on `Begin` or `Commit` the ado transaction call only still have its session. And it must be opened.